test/render-test.js

```diff
@@ -1,72 +1,27 @@
 require('./helper');
 
-var fs = require('fs');
-var path = require('path');
-var _files = path.join(__dirname, '_files');
+var renderHelper = require('./render-helper');
 
-function getContents(testName, ext) {
-  try {
-    return fs.readFileSync(path.join(_files, testName + '.' + ext), 'utf8');
-  } catch (ex) {
-    return null;
-  }
-}
-
-function getView(testName) {
-  var view = getContents(testName, 'js');
-  if (!view) throw new Error('Cannot find view for test "' + testName + '"');
-  return eval(view);
-}
-
-function getPartial(testName) {
-  try {
-    return getContents(testName, 'partial');
-  } catch (error) {
-    // No big deal. Not all tests need to test partial support.
-  }
-}
-
-function getTest(testName) {
-  var test = {};
-  test.view = getView(testName);
-  test.template = getContents(testName, 'mustache');
-  test.partial = getPartial(testName);
-  test.expect = getContents(testName, 'txt');
-  return test;
-}
-
-// You can put the name of a specific test to run in the TEST environment
-// variable (e.g. TEST=backslashes vows test/render-test.js)
-var testToRun = process.env.TEST;
-
-var testNames;
-if (testToRun) {
-  testNames = [testToRun];
-} else {
-  testNames = fs.readdirSync(_files).filter(function (file) {
-    return (/\.js$/).test(file);
-  }).map(function (file) {
-    return path.basename(file).replace(/\.js$/, '');
-  });
-}
+var tests = renderHelper.getTests();
 
 describe('Mustache.render', function () {
   beforeEach(function () {
     Mustache.clearCache();
   });
 
-  testNames.forEach(function (testName) {
-    var test = getTest(testName);
+  tests.forEach(function (test) {
+    var view = eval(test.view);
```

Comment:
Okey, lets say we do `eval` directly in `renderHelper.getView()`. For node it works perfectly :+1: Although for in browsers we now suddenly got 15 broken tests. Why is that? Example from 1 of the 15 broken tests: `./test/_files/escaped.js`:

```javascript
({
  title: function () {
    return "Bear > Shark";
  },
  entities: "&quot; \"'<>/"
})
```

`eval`ing that view in node is perfectly fine; it evaluates the view containing the function and all, ready to be invoked by the current process. Although for browsers its complete different because all the render tests, views and expected output has to be inlined ahead of time as browsers cant read the tests files we have in `./test/_files`. Browsers therefore need the raw views as string so they can `eval` them upon executing the tests on saucelabs which is a completely different process than the one reading `./test/_files`.

Did that clearify something at all? :smile: